app/features-json/setup_json_controller.rb

```diff
@@ -8,5 +8,19 @@ class SetupJSONController < APIController
     get "#{HOME}/configured", authenticate: false do
       return json(Services.onboarding_service.correct_setup?)
     end
+
+    get "#{HOME}/user_details" do
+      code_hosting_service = current_user_config_service.code_hosting_service(
+        provider_credential: Services.provider_credential
```

Comment:
`Services.provider_credential` is the wrong credential still, gotta get the current user.

Reply:
The API here is weird. if a `code_hosting_service` is coming back with credentials of the current user, even though you passed in a `Services.provider_credential`, that's broken. I think what could be happening is that your current user is the same user as your bot.